src/common/utils/fhir/formatter.ts

```diff
@@ -0,0 +1,48 @@
+import { ProductDto } from 'src/common/dtos/product.dto';
+import { SystemProductDto } from 'src/common/dtos/system-product.dto';
+import { R4 } from '@ahryman40k/ts-fhir-types';
+
+export const getSystems = (products: ProductDto[]): Array<any> => {
+  const mappings = products.map(product => product.mappings);
+  const m = [].concat.apply([], mappings);
+  return [...new Set(m.map(m => m.systemName))];
+};
+
+export const formatProductsToFhir = (
```

Comment:
The returned data from this function seems to be missing key attributes like resource url, identifier etc. The code is great. We just need to discuss the attributes, interpretation and so on.

Reply:
those are optional attributes but if we have them then we can easily add them to the returned data.

Reply:
Okay. We should setup a call and you can walk me through your understanding of the things. Mine is not that strong.

Reply:
yes I think we should have a call,